service/src/main/java/com/derpgroup/quip/manager/QuipManager.java

```diff
@@ -249,9 +249,127 @@ protected void doConversationRequest(VoiceInput voiceInput, SsmlDocumentBuilder
     case "HELP":
       doHelpRequest(voiceInput, builder);
       break;
+    case "WHO_BUILT_YOU":
+      doWhoBuiltYouRequest(voiceInput, builder);
+      break;
+    case "WHAT_DO_YOU_DO":
+      doWhatDoYouDoRequest(voiceInput, builder);
+      break;
+    case "FRIENDS":
+      doFriendsRequest(voiceInput, builder);
+      break;
+    case "WHO_IS":
+      doWhoIsRequest(voiceInput, builder);
+      break;
     default:
       builder.text("Unknown request type '" + messageSubject + "'.");
     }
   }
 
+  private void doWhoIsRequest(VoiceInput voiceInput, SsmlDocumentBuilder builder) {
+    String bot = (String) voiceInput.getMetadata().get("bot");
+    String botInQuestion = voiceInput.getMessageAsMap().get("botName");
+    if(bot == null || bot.isEmpty() || botInQuestion == null || botInQuestion.isEmpty()){
+      builder.text("I don't have any info for this situation.");
+      return;
+    }
+    switch(bot){
+    case "complibot":
+      if(botInQuestion.equals(bot)){
+        builder.text("That's me!  ").endSentence();
+        doWhatDoYouDoRequest(voiceInput, builder);
+      }else if(botInQuestion.equals("insultibot")){
```

Comment:
The java-ism would be to put the string constant on the LHS.
`} else if ("insultibot".equals(botInQuestion)) {`

I prefer to use StringUtils:
`} else if (StringUtils.equals(botInQuestion, "insultibot"))`

Also, "insultibot" should probably be defined in a constant somewhere.